swagger-backend.yaml

```diff
@@ -0,0 +1,191 @@
+---
+swagger: '2.0'
+info:
+  description: Flotta backend API for remote data storage
+  title: Flotta backend API
+  version: 1.0.0
+schemes:
+  - https
+consumes:
+  - application/json
+produces:
+  - application/json
+basePath: /api/flotta-backend/v1
+definitions:
+  error:
+    type: object
+    required:
+      - message
+    properties:
+      message:
+        type: string
+  device-configuration-response:
+    type: object
+    properties:
+      message:
+        type: string
+      device-configuration:
+        $ref: 'swagger.yaml#/definitions/device-configuration'
+  device-registration-request:
+    type: object
+    properties:
+      status:
+        type: boolean
+  device-registration-response:
+    type: object
+    properties:
+      target-namespace:
+        type: string
+      message:
+        type: string
+paths:
+  '/namespaces/{namespace}/devices/{device-id}/deregister':
```

Comment:
I don't think that `namespaces` path segment is needed.

Reply:
Or maybe general path pattern like `/devices/{namespace}/{device-id}/...`?

Reply:
that would be a bit confusing, as the `{namespace}` could be interepreted as being the `{device-id}` and vice versa. I'd prefer to either remove the namespace and put it as part of the payload or keep it as it is.

Reply:
OK. That's fine.